File: src/test/java/org/dxworks/githubminer/service/repository/branches/GithubBranchServiceIT.java
package org.dxworks.githubminer.service.repository.branches;

import org.dxworks.githubminer.dto.response.repository.branches.Branch;
import org.dxworks.githubminer.utils.TestUtils;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class GithubBranchServiceIT {

    private GithubBranchService githubBranchService = new GithubBranchService("MarioRivis", "SimpleRegistrationExample", TestUtils.getGithubCredentials());

    @Test
    void tetGetAllBranches() {
        List<Branch> allCommits = githubBranchService.getAllBranches();
        assertTrue(allCommits.stream().allMatch(repoCommit -> repoCommit instanceof Branch));
        assertEquals(8, allCommits.size());
    }
}

File: github-miner-core/src/test/kotlin/org/dxworks/githubminer/service/repository/branches/GithubBranchServiceIT.kt
package org.dxworks.githubminer.service.repository.branches

import org.dxworks.githubminer.dto.response.repository.branches.Branch
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class GithubBranchServiceIT {
    private val githubBranchService = GithubBranchService("apache", "kafka")

    @Test
    fun tetGetAllBranches() {
        val allCommits = githubBranchService.allBranches
        Assertions.assertTrue(allCommits.stream().allMatch { repoCommit: Branch? -> repoCommit is Branch })
        Assertions.assertEquals(9, allCommits.size)
    }
}
